Jellyfin.Server/Migrations/MigrationRunner.cs

```diff
@@ -111,38 +119,60 @@ private static void HandleStartupWizardCondition(IEnumerable<IMigrationRoutine>
             migrationOptions.Applied.AddRange(onlyOldInstalls.Select(m => (m.Id, m.Name)));
         }
 
-        private static void PerformMigrations(IMigrationRoutine[] migrations, MigrationOptions migrationOptions, Action<MigrationOptions> saveConfiguration, ILogger logger)
+        private static async Task PerformMigrations(
+            IMigrationRoutine[] migrations,
+            MigrationOptions migrationOptions,
+            Action<MigrationOptions> saveConfiguration,
+            ILogger logger,
+            IJellyfinDatabaseProvider? jellyfinDatabaseProvider)
         {
             // save already applied migrations, and skip them thereafter
             saveConfiguration(migrationOptions);
             var appliedMigrationIds = migrationOptions.Applied.Select(m => m.Id).ToHashSet();
+            var migrationsToBeApplied = migrations.Where(e => !appliedMigrationIds.Contains(e.Id)).ToArray();
 
-            for (var i = 0; i < migrations.Length; i++)
+            string? migrationKey = null;
+            if (migrationsToBeApplied.Any(f => f is IDatabaseMigrationRoutine) && jellyfinDatabaseProvider is not null)
```

Comment:
Lets do the cheaper check first
```suggestion
            if (jellyfinDatabaseProvider is not null && migrationsToBeApplied.Any(f => f is IDatabaseMigrationRoutine))
```

Reply:
Technically this should never happen, only if somebody were to try to perform a database related migration on the pre-migration stage but i guess you are right